cachito/web/models.py

```diff
@@ -186,6 +186,30 @@ def to_purl(self):
             # see https://github.com/package-url/purl-spec/issues/63 for further reference
             purl_name = self.name.replace("/", "%2F")
             return f"pkg:golang/{purl_name}@{self.version}"
+
+        elif self.type == "npm":
+            purl_name = self.name.replace("@", "%40")
+            match = re.match(r"(?P<protocol>[^:]+):(?://)?(?P<suffix>.+)", self.version)
```

Comment:
```suggestion
            match = re.match(r"^(?P<protocol>[^:]+):(?://)?(?P<suffix>.+)$", self.version)
```

Reply:
`re.match` already starts the match from the beginning of the string. I am wondering if there are any benefits in saying you want to match the end of a string when searching for `.+`.

Reply:
Ah, you're right. I didn't know that `re.match` did that already.